src/ldp.rs

```diff
@@ -53,6 +53,120 @@ lazy_static! {
     };
 }
 
+pub fn get_proof_suite(proof_type: &str) -> Result<&(dyn ProofSuite + Sync), Error> {
```

Comment:
What problem making `type_` an enum would incur? Inability to deserialise/manipulate documents with unsupported proof types? What if we had an `Other(string)` untagged variant, is that possible? Could we still keep the differentiation `NotImplemented` vs `NotEnabled`?

Reply:
A potential problem with `type_` as an enum would be for extensibility - although it might not be worse than we have currently have (before or after this PR). For #174 we may want to allow defining proof type outside ssi itself - like how the DID methods/resolvers are implemented.